dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+
+    while request is not None:
+      response = request.execute()
+
+      for zone in response.get('items',{}).values():
+
+        for disk in zone.get('disks', {}):
+          resource = Resource()
+          resource.id = disk.get('id')
+          resource.resource_name = disk.get('selfLink')
+          resource.creation_timestamp = disk.get('creationTimestamp')
+          if disk.get('sourceDisk'):
+            resource.parent = Resource()
+            resource.parent.resource_name = disk.get('sourceDisk')
+            resource.parent.id = disk.get('sourceDiskId')
+          elif disk.get('sourceSnapshot'):
+            resource.parent = Resource()
+            resource.parent.resource_name = disk.get('sourceSnapshot')
+            resource.parent.id = disk.get('sourceSnapshotId')
+          elif disk.get('sourceImage'):
+            resource.parent = Resource()
+            resource.parent.resource_name = disk.get('sourceImage')
+            resource.parent.id = disk.get('sourceImageId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.disks().aggregatedList_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfDiskImages(self, project_id: str) -> Dict[str, Resource]:
+    """Retrieves list of disk images in a project.
+
+    Args:
+      project_id: Project Id to retrieve the list of disk images for
+
+    Return:
+      Dict of disk images
+    """
+    result: Dict[str, Resource] = {}
+
+    # Using beta version of the API because v1 did not have important
+    # information when creating this script
+    compute_api_client = CreateService('compute', 'beta')
+
+    # Disk images are not tied to a zone, so there is no aggregatedList
+    request = compute_api_client.images().list(project=project_id)  # pylint: disable=no-member
+
+    while request is not None:
+      response = request.execute()
+
+      if response:
+        for image in response.get('items', {}):
+          resource = Resource()
+          resource.id = image.get('id')
+          resource.resource_name = image.get('selfLink')
+          resource.creation_timestamp = image.get('creationTimestamp')
+          if image.get('sourceDisk'):
+            resource.parent = Resource()
+            resource.parent.resource_name = image.get('sourceDisk')
+            resource.parent.id = image.get('sourceDiskId')
+          elif image.get('sourceSnapshot'):
+            resource.parent = Resource()
+            resource.parent.resource_name = image.get('sourceSnapshot')
+            resource.parent.id = image.get('sourceSnapshotId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.images().list_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfSnapshots(self, project_id: str) -> Dict[str, Resource]:
```

Comment:
same as _RetrieveListOfDiskImages

Reply:
libcloudforensics doesn't have the ability to list snapshots